Sources/TSCBasic/FileSystem.swift

```diff
@@ -1159,8 +1159,19 @@ public class RerootedFileSystemView: FileSystem {
     }
 }
 
+private var _localFileSystem: any FileSystem = LocalFileSystem()
+
 /// Public access to the local FS proxy.
-public var localFileSystem: FileSystem = LocalFileSystem()
+public var localFileSystem: any FileSystem {
+    get {
+         return _localFileSystem
+    }
+
+    @available(*, deprecated, message: "This global should never be mutable and is supposed to be read-only")
```

Comment:
add a date here in a comment so we know when this was first deprecated and get delete once enough time has passed